taskcat/_config.py

```diff
@@ -131,6 +147,92 @@ def _get_templates(self):
                 client_factory_instance=test.client_factory,
             )
 
+    def _assign_regional_factories(self):
+        for test_name, test_obj in self.tests.items():
+            for region in test_obj.regions:
+                self._set_appropriate_creds(test_name, region)
+
+    def _set_appropriate_creds(self, test_name, region):
+        if hasattr(region.client, "set") and region.client.set:
+            return
+        cred_key_list = [
+            "default",
+            f"{test_name}_default",
+            region.name,
+            f"{test_name}_{region.name}",
+        ]
+        for cred_key in cred_key_list:
+            client_factory = self._client_factory_instance.return_credset_instance(
+                cred_key
+            )
+            if not client_factory:
+                continue
+            region.client = client_factory
+            region.client.set = True
+            sts_client = region.client.get("sts")
+            try:
+                account = sts_client.get_caller_identity()["Account"]
+            except ClientError as e:
```

Comment:
can we also trap `botocore.exceptions.NoCredentialsError: Unable to locate credentials` I ran into this by accident as I had disabled my default creds

Reply:
?